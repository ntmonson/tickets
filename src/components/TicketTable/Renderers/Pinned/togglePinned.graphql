mutation toggleTicketPinned($id: String!) {
  databaseToggleTicketPinned(id: $id) {
    id
    topic
    closed
    pinned
    createdAt
    updatedAt
  }
}
