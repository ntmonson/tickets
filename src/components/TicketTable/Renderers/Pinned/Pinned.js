import React from 'react';
import PropTypes from 'prop-types';
import { graphql, compose } from 'react-apollo';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import togglePinned from './togglePinned.graphql';
import s from './Pinned.css';

const Pointer = ({ value, ticketId, togglePin }) => {
  const color = value ? 'Gold' : '';
  return (
    <button style={{ color }} onClick={() => togglePin(ticketId)}>
      <i className="fas fa-thumbtack" />
    </button>
  );
};

Pointer.propTypes = {
  value: PropTypes.bool.isRequired,
  ticketId: PropTypes.string.isRequired,
  togglePin: PropTypes.func.isRequired,
};

const graphqlQueries = graphql(togglePinned, {
  name: 'togglePin',
  props: ({ togglePin }) => ({
    togglePin: id => togglePin({ variables: { id } }),
  }),
});

export default compose(withStyles(s), graphqlQueries)(Pointer);
