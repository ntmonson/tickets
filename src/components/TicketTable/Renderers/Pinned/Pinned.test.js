/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import { Pinned } from './Pinned';

describe('Pinned', () => {
  test('Renders Pinned component correctly with value 1', () => {
    const component = renderer
      .create(<Pinned value={1} ticketId="asdf1234" togglePin={() => {}} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('Renders Pinned component correctly with value 0', () => {
    const component = renderer
      .create(<Pinned value={0} ticketId="asdf1234" togglePin={() => {}} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
