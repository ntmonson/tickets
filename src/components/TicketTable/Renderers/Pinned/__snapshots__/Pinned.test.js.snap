// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pinned Renders Pinned component correctly with value 0 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "color": "",
    }
  }
>
  <i
    className="fas fa-thumbtack"
  />
</button>
`;

exports[`Pinned Renders Pinned component correctly with value 1 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "color": "Gold",
    }
  }
>
  <i
    className="fas fa-thumbtack"
  />
</button>
`;
